src/object/cli_shard.c

```diff
@@ -1310,6 +1370,7 @@ dc_obj_shard_rw(struct dc_obj_shard *shard, enum obj_rpc_opc opc,
 				D_ERROR("crt_req_set_timeout error: %d", rc);
 		    }
 
+		dc_obj_metrics_incr_inflightcntr(req->cr_opc);
```

Comment:
Hmm, these incr_completecntr() can even be moved to daos_rpc_cb(), instead of called everywhere?

Reply:
inflightcntr is never decreased. However, when I report the inflight count to the end user, I do a diff between inflight and (success + failure). Please refer dc_metrics_cntr_copy(). I agree with you to change the name to sendcntr instead of inflightcntr. I will make this change.

incr_completecntr() updates success or failure counter based on the return value of the rpc call. Since the success or failure cannot be determined at daos_rpc_cb() I moved this to the end of cb of each rpc call.

Reply:
ah, ok. thanks